server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestCloseIndexAction.java

```diff
@@ -55,16 +53,7 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         closeIndexRequest.indicesOptions(IndicesOptions.fromRequest(request, closeIndexRequest.indicesOptions()));
         String waitForActiveShards = request.param("wait_for_active_shards");
         if ("index-setting".equalsIgnoreCase(waitForActiveShards)) {
-            deprecationLogger.warn(
-                DeprecationCategory.SETTINGS,
-                "close-index-wait_for_active_shards-index-setting",
-                "?wait_for_active_shards=index-setting is now the default behaviour; the 'index-setting' value for this parameter "
-                    + "should no longer be used since it will become unsupported in version "
-                    + (Version.V_8_0_0.major + 1)
-            );
-            // TODO in v9:
-            // - throw an IllegalArgumentException here
-            // - record the removal of support for this value as a breaking change
+            throw new IllegalArgumentException("wait_for_active_shards=index-setting shouldn't be used");
```

Comment:
We need to preserve the v8 behaviour here still if `RestApiVersion.V_8` is specified.

Reply:
:+1: that applies to all other similar breaking changes too btw